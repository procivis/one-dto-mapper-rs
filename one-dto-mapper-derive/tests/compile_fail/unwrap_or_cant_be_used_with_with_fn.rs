use one_dto_mapper_derive::From;

struct OptionalDto {
    age: Option<u16>,
}

#[derive(From)]
#[from(NumericDto)]
struct StringDto {
    #[from(unwrap_or = "16")]
    #[from(with_fn = "std::convert::identity")]
    age: u16,
}

fn main() {}
