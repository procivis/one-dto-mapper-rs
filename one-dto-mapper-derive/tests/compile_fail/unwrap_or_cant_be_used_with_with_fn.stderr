error: `unwrap_or` cannot be used together with `with_fn` or `with_fn_ref` at age
 --> tests/compile_fail/unwrap_or_cant_be_used_with_with_fn.rs:7:10
  |
7 | #[derive(From)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `From` (in Nightly builds, run with -Z macro-backtrace for more info)
